visualization/app/codeCharta/codeCharta.component.ts

```diff
@@ -42,20 +42,19 @@ export class CodeChartaController {
 		this.loadFileOrSample()
 	}
 
-	public loadFileOrSample() {
-		return this.urlUtils
-			.getFileDataFromQueryParam()
-			.then((data: NameDataPair[]) => {
-				if (data.length > 0) {
-					this.tryLoadingFiles(data)
-					this.setRenderStateFromUrl()
-				} else {
-					this.tryLoadingSampleFiles()
-				}
-			})
-			.catch(() => {
+	async loadFileOrSample() {
+		try {
+			const data = await this.urlUtils.getFileDataFromQueryParam()
+			if (data.length > 0) {
+				this.tryLoadingFiles(data)
+				this.setRenderStateFromUrl()
+			} else {
 				this.tryLoadingSampleFiles()
-			})
+			}
+		} catch (error) {
+			console.error(error)
```

Comment:
We should probably improve the error handling here?

Reply:
I assume, that his piece of code is quite old and from a time, where we had no proper validation.
I'm not sure if this catch-block is ever reached. We catch the errros already during validation.

Reply:
We do trigger it in our tests. We likely won't reach this point in "regular" code though.

Reply:
It is actually possible to reach this in regular code. This part is there to load the files and there could e.g., be a network outage.

Our error handling here indicates that there was an error but it does not provide any information about what actually has happened. We might want to change that.

Reply:
How do we determine the error though? Should we throw something generic like `Something went wrong while loading files. Please try again or contact us on Github`?

Reply:
That is already the case. The error is: `One or more files from the given file URL parameter could not be loaded. Loading sample files instead.`. I thought we might extend it to: `One or more files from the given file URL parameter could not be loaded. Loading sample files instead.\n\nError: $error.message`

Reply:
e2e test is not happy atm